sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.service.ts

```diff
@@ -0,0 +1,161 @@
+  constructor(
+    @InjectRepository(Student)
+    private readonly studentRepo: Repository<Student>,
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+    @InjectRepository(SFASBridgeLog)
+    private readonly sfasBridgeLogRepo: Repository<SFASBridgeLog>,
+  ) {}
+
+  /**
+   * Get the latest bridge file log date.
+   * When there is no log, null will be returned.
+   * @returns latest bridge file log date.
+   */
+  async getLatestBridgeFileLogDate(): Promise<Date | null> {
+    const [latestBridgeFileLog] = await this.sfasBridgeLogRepo.find({
+      select: { id: true, referenceDate: true },
+      order: { referenceDate: "DESC" },
+      take: 1,
+    });
+    return latestBridgeFileLog ? latestBridgeFileLog.referenceDate : null;
+  }
+
+  /**
+   * Log the details of bridge file that was sent to SFAS.
+   * @param referenceDate date when the bridge file data was extracted.
+   * @param fileName bridge file name.
+   */
+  async logBridgeFileDetails(
+    referenceDate: Date,
+    fileName: string,
+  ): Promise<void> {
+    await this.sfasBridgeLogRepo.insert({
+      referenceDate,
+      generatedFileName: fileName,
+    });
+  }
+
+  /**
+   * Get all student ids of students who have one or more updates
+   * since the date provided.
+   * The updates can be one or more of the following:
+   * - Student or User data
+   * - Sin validation data
+   * - Cas supplier data
+   * - Overawards data
+   * @param modifiedSince the date after which the student data was updated.
+   * If not provided, all the students with at least one submitted application will be returned.
+   */
+  async getAllStudentsWithUpdates(modifiedSince?: Date): Promise<number[]> {
+    const applicationStudentQuery = this.applicationRepo
+      .createQueryBuilder("application")
+      .select(["application.id", "student.id"])
+      .distinctOn(["student.id"])
+      .innerJoin("application.student", "student")
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
+      .innerJoin("student.casSupplier", "casSupplier")
+      .leftJoin("student.overawards", "overaward")
+      .where("application.applicationStatus != :overwritten", {
+        overwritten: ApplicationStatus.Overwritten,
+      })
+      .andWhere("application.currentAssessment is not null");
+    if (modifiedSince) {
+      applicationStudentQuery
+        .andWhere(
+          new Brackets((qb) => {
+            qb.where("student.updatedAt > :modifiedSince")
+              .orWhere("user.updatedAt > :modifiedSince")
+              .orWhere("sinValidation.updatedAt > :modifiedSince")
+              .orWhere("casSupplier.updatedAt > :modifiedSince")
+              .orWhere("overaward.updatedAt > :modifiedSince");
+          }),
+        )
+        .setParameter("modifiedSince", modifiedSince);
+    }
+    const applicationsWithStudentUpdates =
+      await applicationStudentQuery.getMany();
```

Comment:
Maybe using the `getRawMany` and selecting only the student's ids would be better to avoid typeorm parsing all the returned results into objects that will not be needed.

Reply:
I tried the `getRawMany()` option, the SQL seems to be executed fine but when typeorm isn't able to read the values without error. 

![image](https://github.com/user-attachments/assets/d2b725cd-9189-4629-980a-2efb981ba7b2)

Reply:
Can you please try the below?
```ts
.createQueryBuilder("application")
.select("student.id", "studentId")
.distinctOn(["studentId"])
.innerJoin("application.student", "student")
.innerJoin("student.user", "user")
.innerJoin("student.sinValidation", "sinValidation")
.innerJoin("student.casSupplier", "casSupplier")
.leftJoin("student.overawards", "overaward")
.where("application.applicationStatus != :overwritten")
```

Reply:
I did try that as well. With `(["student.id"])` and `("student.id", "studentId")`

Reply:
Thanks for the nice discussion. 
As we discussed, we figured out that if we are using an alias name to perform an operation which in this case is `distinct on` we must enclose the alias within `"` e.g. `"studentId"` otherwise it doesn't work in typeorm. 
We followed the same approach to have the `getRawMany` working.